### ISSUE TITLE
Chat Window Grows Steadily And Doesn't Scroll Automatically

### ISSUE THREAD

**Issue by MagnificentJake** [sPOiDar/fvtt-module-stream-view#2]
When starting empty, the chat window is sized for ~3 die roll cards. Die rolls beyond this cause the chat window to scale to full vertical height, but also introduce a scroll bar that means the chat window is always one roll behind, meaning you have to go into the stream view and scroll down.

(This is a great module and I'm really appreciative that it exists!)

**Comment by sPOiDar** [comment]
Hrm, I can't reproduce - for me the chat log scrolls automatically, and the most recent roll remains pinned to the bottom of the popout. Do you have any other modules that affect the chat log at all? I'm happy to look into it, but I need to be able to reproduce the behaviour you're seeing.

**Comment by MagnificentJake** [comment]
Ah, a bit of trial and error suggests it's an issue with Dice So Nice – if that module isn't set to Display Chat Message Immediately the delay between rolling the die and the chat message showing up seems to interfere with the scrolling. It's obviously an easy workaround just to turn that setting on though!

**Comment by sPOiDar** [comment]
Interesting, I'll try to find some time to have a play around with DSN and see if I can find why it does that, as I'm just using the standard chat log popout, but with the header and input parts removed.

**Comment by sPOiDar** [comment]
Turns out this is a bug in core Foundry - when chat is in a popout, calling `ui.chat.scrollBottom()` (which is how DSN and probably some other modules scroll the chat log after appending a message) does not target the popout chat, only the original one in the sidebar.

I've reported via Discord, so I'm going to close this issue - I expect a future Foundry release will resolve this for good. If for some unexpected reason they decline to solve it there, I'll re-open and implement a fix to core in this module.
